### PR TITLE
Post migration andre

### PR DESCRIPTION
**Pull request by atGit2021**
Daniel, please review the PgPostRepository and post.module file.  I added the create and readOne functionality for posts. Thank you, Andre

┆Issue is synchronized with this [Monday item](https://seed-company-squad.monday.com/boards/3451697530/pulses/3466559176) by [Unito](https://www.unito.io)

### CODE REVIEW

File: src/components/post/post.module.ts [hunk 1]
```diff
@@ -20,6 +22,7 @@ import { PostableResolver } from './postable.resolver';
     PostableResolver,
     PostLoader,
     ...Object.values(migrations),
+    splitDb(PostRepository, PgPostRepository),
```
Comment (CarsonF) [inline src/components/post/post.module.ts:25]: This should replace line 21.

File: src/components/post/post.pg.respository.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    const [id] = await this.pg.query<{ id: ID }>(
+      `
+      INSERT INTO sc.posts(type, shareability, body, created_by, modified_by, owning_person, owning_group)
+      VALUES($1, $2, $3, $4, 
+          (SELECT person FROM admin.tokens WHERE token = 'public'), 
+          (SELECT person FROM admin.tokens WHERE token = 'public'), 
+          (SELECT id FROM admin.groups WHERE  name = 'Administrators'))
+      RETURNING id;
+      `,
+      [input.type, input.shareability, input.body, session.userId]
```
Comment (daniel-leod) [inline src/components/post/post.pg.respository.ts]: Please omit the `session.userId` parameter.

**Review by daniel-leod — CHANGES_REQUESTED** [review]

**Review by CarsonF — CHANGES_REQUESTED** [review]

File: src/components/post/post.pg.respository.ts [hunk 3]
```diff
@@ -1,33 +1,57 @@
...
       throw new ServerException('Failed to create post');
     }
 
     return id;
   }
+
+  async readOne(id: ID): Promise<UnsecuredDto<Post>> {
+    const rows = await this.pg.query<UnsecuredDto<Post>>(
+      `
+      SELECT id, directory, type, shareability, body, created_at as "createdAt", modified_at as "modifiedAt"
```
Comment (daniel-leod) [inline src/components/post/post.pg.respository.ts:44]: To follow the same structure as `post.dto` we could omit the `directory` field and cast the `created_by` as `creator`.

Comment (daniel-leod) [reply src/components/post/post.pg.respository.ts:44]:
```suggestion
      SELECT id, created_by as "creator", type, shareability, body, created_at as "createdAt", modified_at as "modifiedAt"
```

**Review by daniel-leod — CHANGES_REQUESTED** [review]

File: src/components/post/post.pg.repository.ts [hunk 4]
```diff
@@ -54,4 +54,9 @@ export class PgPostRepository extends PostRepository {
 
     return rows[0];
   }
+
+  @PgTransaction()
```
Comment (daniel-leod) [inline src/components/post/post.pg.repository.ts:58]: Please omit this decorator. We use `@PgTransaction()` when it's needed to run multiple queries in a specific order

Comment (daniel-leod) [reply src/components/post/post.pg.repository.ts:58]:
```suggestion

```

**Review by daniel-leod — CHANGES_REQUESTED** [review]

File: src/components/post/post.pg.repository.ts (hunk 4 above)
Comment (CarsonF) [reply src/components/post/post.pg.repository.ts:58]: It shouldn't hurt anything to include it.

Comment (daniel-leod) [reply src/components/post/post.pg.repository.ts:58]: You're right, we can leave it as it is.

File: src/components/post/post.pg.respository.ts (hunk 3 above)
Comment (atGit2021) [reply src/components/post/post.pg.respository.ts]: All set, ready for review

**Comment by CarsonF** [comment]
Closing postgres migration PRs for now until a more wholistic strategy/plan can be formed.

**Pull request closed without merging**